notes/12-guarding-routes.md

```diff
@@ -0,0 +1,195 @@
+# Guarding Routes
+
+In this exercise, we will be adding redirection in the UI flow based on the availability of an authenticated session.
+Specifically, users should be able to access the `login` route, only when an authenticated session _does not exist_. And similarly, users should be able to access the `teams` route only when an authenticated _session exists_.
+
+So let's get started.
+
+## Changes Needed in Javascript
```

Comment:
I think we should (1) make this title more descriptive and (2) break the exercise up into discrete "tasks" 

- "Enhancing the `auth` service"
- "Adding a redirect to the `/teams` page"
- "Adding a redirect to the `/login` page"